sdk/storage/azure-storage-data-movement/src/main/java/com/azure/storage/data/movement/PathScanner.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.data.movement;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.BaseStream;
+
+public class PathScanner {
+    private final ClientLogger logger = new ClientLogger(PathScanner.class);
+    private final String basePath;
+
+    public PathScanner(String path) {
+        try {
+            // Resolve the given path to an absolute path in case it isn't one already
+            Path pathObj = Paths.get(path);
+            basePath = pathObj.toAbsolutePath().toString();
+
+            // Check if the path exists and whether or not it's a directory; throw
+            // an error if there's nothing present or readable at the given path
+            if (!(Files.isDirectory(pathObj) || Files.isRegularFile(pathObj))) {
+                throw new IllegalArgumentException(String.format("No item(s) located at the path '%s'.", basePath));
```

Comment:
Wouldn't a better error message be that the passed path isn't a folder? This error is a bit misleading as you could have a directory, non-regular file path that doesn't contain any items.

Reply:
Ah, the comment I have is a bit misleading because the code following is changed, I need to update that. This no longer checks whether it's a file or directory, as that logic is handled in the scan itself; instead, it solely checks whether or not the given path exists on the local system. Files::exists had the caveat of also returning false if the file/folder was unreadable, which might not be optimal to check before the .scan() call, so instead ctor just looks for either isDir or isFile to be true.